model_compression_toolkit/quantizers_infrastructure/pytorch/quantize_wrapper.py

```diff
@@ -41,17 +44,53 @@ def __init__(self,
                 dispatcher: A node quantization dispatcher.
             """
             super().__init__()
-            self.dispatcher = dispatcher
+            self._dispatcher = dispatcher
             if isinstance(module, nn.Module):
                 self.add_module(LAYER, module)
             else:
                 # Functional layers
                 setattr(self, LAYER, module)
 
+            # Init weight quantizers from dispatcher
+            self._set_weight_quantizers()
+
+            # Init activations quantizers from dispatcher
+            self._set_activation_quantizers()
+
+        def convert_to_inferable_quantizers(self):
+            # Activation quantizers
+            if self._dispatcher.is_activation_quantization:
+                inferable_activation_quantizers = []
+                for quantizer in self._dispatcher.activation_quantizers:
+                    if isinstance(quantizer, qi.BasePytorchTrainableQuantizer):
+                        inferable_activation_quantizers.append(quantizer.convert2inferable())
+                    else:
+                        Logger.error('Can only convert trainable quantizers based on BasePytorchTrainableQuantizer')
+                self.set_activation_quantizers(inferable_activation_quantizers)
+
+            # Weight quantizers
+            if self._dispatcher.is_weights_quantization:
+                inferable_weight_quantizers = {}
+                for name, quantizer in self._dispatcher.weight_quantizers.items():
+                    if isinstance(quantizer, qi.BasePytorchTrainableQuantizer):
+                        inferable_weight_quantizers.update({name: quantizer.convert2inferable()})
+                    else:
+                        Logger.error('Can only convert trainable quantizers based on BasePytorchTrainableQuantizer')
+                self.set_weight_quantizers(inferable_weight_quantizers)
+
+        def set_weight_quantizers(self, weight_quantizers: List[BaseInferableQuantizer]):
+            self._dispatcher.set_weight_quantizers(weight_quantizers)
+            self._set_weight_quantizers()
+
+        def set_activation_quantizers(self, activation_quantizers: List[BaseInferableQuantizer]):
+            self._dispatcher.set_activation_quantizers(activation_quantizers)
+            self._set_activation_quantizers()
+
+        def _set_weight_quantizers(self):
             self._weight_vars = []
 
             # Init weights quantizers
-            for name, quantizer in self.dispatcher.weight_quantizers.items():
+            for name, quantizer in self._dispatcher.weight_quantizers.items():
```

Comment:
This mean you run "initialize_quantization" and  "weight.detach()" to inferable quantizers